runelite-client/src/main/java/net/runelite/client/plugins/microbot/tutorialisland/TutorialIslandScript.java

```diff
@@ -332,20 +342,29 @@ public void CombatGuide() {
             if (isInDialogue()) return;
             if (Microbot.getVarbitPlayerValue(281) == 500) {
                 Rs2Walker.walkTo(new WorldPoint(3111, 9526, Rs2Player.getWorldLocation().getPlane()));
-                Rs2GameObject.interact("Ladder","Climb-up");
+                sleep(5000);
+                sleepUntil(() -> !Rs2Player.isMoving());
+                System.out.println("We're done moving...");
+                Rs2GameObject.interact(9727,"Climb-up");
                 sleepUntil(() -> Microbot.getVarbitPlayerValue(281) != 500);
+                System.out.println("Sleeping varbit");
                 return;
             }
             if (Microbot.getVarbitPlayerValue(281) == 480 || Microbot.getVarbitPlayerValue(281) == 490) { // killl rat with range
                 Actor rat = Microbot.getClient().getLocalPlayer().getInteracting();
                 if (rat != null && rat.getName().equalsIgnoreCase("giant rat")) return;
                 Rs2Inventory.wield( "Shortbow");
+                sleep(1000,1200);
                 Rs2Inventory.wield("Bronze arrow");
+                sleep(2000,2500);
                 Rs2Npc.attack("Giant rat");
                 return;
             }
             if (Microbot.getVarbitPlayerValue(281) == 470) {
-                Rs2Walker.walkTo(new WorldPoint(3109, 9511, 0));
+                // If we don't have a particular point, we can be on the other
+                // side of the gate and try to talk to him infinitely!!
+                WorldPoint combatTutorLoc = new WorldPoint(3109,9511,0);
```

Comment:
webwalker default is 10 tiles for finish. we have to manually move to vanaka because we can "finish" on the rat-side of the gate. moving will open the gate and find us on the correct side.